aws-fargate/miniwdl_aws_fargate.py

```diff
@@ -0,0 +1,208 @@
+        if self.cfg["task_runtime"].get_bool("as_user"):
+            user = f"{os.geteuid()}:{os.getegid()}"
+
+        fargate_mem_value = self.fargate_mem_values[0]
+        if self.cfg.has_option("aws_fargate", "default_memory_mb"):
+            fargate_mem_value = self.cfg["aws_fargate"].get_int("default_memory_mb")
+        if "memory_reservation" in self.runtime_values:
+            for fargate_mem_value in self.fargate_mem_values:
+                if fargate_mem_value * 1024 * 1024 >= self.runtime_values["memory_reservation"]:
+                    break
+            else:
+                logger.warning("Memory reservation exceeds maximum Fargate memory")
+
+        fargate_cpu_value = self.fargate_cpu_values[0]
+        if self.cfg.has_option("aws_fargate", "default_cpu_shares"):
+            fargate_cpu_value = self.cfg["aws_fargate"].get_int("default_cpu_shares")
+        if "cpu" in self.runtime_values:
+            for fargate_cpu_value in self.fargate_cpu_values:
+                if fargate_mem_value > self.fargate_cpu_mem_constraints[fargate_cpu_value]["max"]:
+                    continue
+                if fargate_cpu_value >= self.runtime_values["cpu"] * 1024:
+                    break
+        fargate_mem_value = max(fargate_mem_value, self.fargate_cpu_mem_constraints[fargate_cpu_value]["min"])
+        logger.info("Task mem %s, CPU %s", fargate_mem_value, fargate_cpu_value)
+
+        wd = os.path.join(self.container_dir, "work")
+
+        for pipe_file in ["stdout.txt", "stderr.txt"]:
+            pathlib.Path(os.path.join(self.host_dir, pipe_file)).touch()
+
+        run_args = [
+            "--command", f"cd {wd} && bash ../command 2> >(tee -a ../stderr.txt 1>&2) > >(tee -a ../stdout.txt)",
+            "--security-group", self.efs_security_group,
+            "--volumes", f"{self.efs_id}:{os.path.relpath(self.host_dir, self.efs_mountpoint)}={self.container_dir}",
+            "--fargate-memory", str(fargate_mem_value),
+            "--fargate-cpu", str(fargate_cpu_value)
+        ]
+        # Tags require "opt in to new ID format"
+        # aws.amazon.com/blogs/compute/migrating-your-amazon-ecs-deployment-to-the-new-arn-and-resource-id-format-2/
+        # "--tags", f"miniwdl_run_id={self.run_id}"
+
+        if user:
+            run_args += ["--user", user]
+        task_desc = ecs.run(ecs.run_parser.parse_args(run_args))
+        exit_code = None
+        try:
+            with contextlib.ExitStack() as cleanup:
+                poll_stderr = cleanup.enter_context(
+                    PygtailLogger(
+                        logger,
+                        os.path.join(self.host_dir, "stderr.txt"),
+                        callback=self.stderr_callback,
+                    )
+                )
+
+                # poll for task exit code
+                was_running = False
+                while exit_code is None:
+                    time.sleep(random.uniform(1.0, 2.0))  # spread out work over the GIL
```

Comment:
The streaming log of `stderr.txt` should be easy to add with the PygtailLogger helper.

Reply:
Isn't that already in place on [L199](https://github.com/chanzuckerberg/miniwdl-plugins/pull/4/files#diff-a11433ddf09e75842df613aafdeeaa0dR199)?